src/consumption.cpp

```diff
@@ -875,8 +876,7 @@ static bool eat( item &food, player &you, bool force )
     const bool saprophage = you.has_trait( trait_SAPROPHAGE );
     if( spoiled && !saprophage ) {
         you.add_msg_if_player( m_bad, _( "Ick, this %s doesn't taste so good…" ), food.tname() );
-        if( !you.has_trait( trait_SAPROVORE ) && !you.has_trait( trait_EATDEAD ) &&
-            ( !you.has_bionic( bio_digestion ) || one_in( 3 ) ) ) {
+        if( !you.has_flag( json_flag_IMMUNE_SPOIL ) || one_in( 3 ) ) {
```

Comment:
Should that one-in-three chance still be there?  This nerfs Saprovore and Eater of the Dead to only make you resistant to spoilage rather than immune to it.  Maybe add a RESIST_SPOIL flag and use that for the bio_digestion bionic instead?

Reply:
Good catch, that said I decided to buff bio_digestive instead.  It's a judgement call but the item info already said you could tolerate it and I'm generally in favor in simplifying some of this stuff when possible as its too complicated already IMO.